props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,861 @@
+Reload                    - Click RELOAD.
+
+-TODO: PLI_ON / OFF, CLIP IN etc....
+
+Wavs to use for talking Mode (else Talkie voice):
+- If these are not present, mode.wav will be used for all modes.
+- If no mode.wav either, then Talkie voice speaks selected mode.
+  mdstun.wav
+  mdkill.wav
+  mdauto.wav
+Additionally:
+  quote.wav      // for playing quotes
+  in.wav         // for blaster power off
+  out.wav        // for blaster power on
+  destruct.wav   // for self destruct overload
+  boom.wav       // for self destruct explosion
+These can be used to match a more "plasmatic" stun sound.
+  clipins.wav
+  clipouts.wav
+  reloads.wav
+
+The following are likely best kept in a 'common' folder in your Font Search Path, but of course
+can be specific versions within a font as well.
+Choose a Voice Pack containing all these sounds here: https://fredrik.hubbe.net/lightsaber/sound/
+
+  Battery level wavs:
+    mzero.wav
+    mnum 1-20 wavs, as well as the "tens" thirty.wav - ninety.wav
+    battery.wav    // for accompanying EFFECT_BATTERY_LEVEL
+    battlevl.wav   // for Spoken Battery Level
+    mpercent.wav
+    mpoint.wav
+    mvolts.wav
+  vmbegin.wav    // for Begin Volume Menu
+  vmend.wav      // for End Volume Menu
+  volup.wav      // for increse volume
+  voldown.wav    // for decrease volume
+  volmin.wav     // for minimum volume reached
+  volmax.wav     // for maximum volume reached
+  mfalse.wav     // "disabled" used for sequential quote mode
+  mtrue.wav      // "enabled" used for sequential quote mode
+  mrandom.wav    // "ramdom" used for sequential quote mode
+*/
+
+#ifndef PROPS_BLASTER_BC_H
+#define PROPS_BLASTER_BC_H
+
+#ifndef BLASTER_DEFAULT_MODE
+#define BLASTER_DEFAULT_MODE MODE_KILL
+#endif
+#include "prop_base.h"
+#include "../sound/sound_library.h"
+
+#define PROP_TYPE BlasterBC
+#define PROP_HAS_BULLET_COUNT
+
+EFFECT(clipin);
+EFFECT(clipins);    // s for stun mode
+EFFECT(clipouts);   // s for stun mode
+EFFECT(clipout);
+EFFECT(empty);
+EFFECT(full);
+EFFECT(jam);
+EFFECT(mode);
+EFFECT(plioff);
+EFFECT(plion);
+EFFECT(range);
+EFFECT(reload);
+EFFECT(reloads);    // s for stun mode
+EFFECT(stun);
+EFFECT(unjam);
+EFFECT(mdstun);
+EFFECT(mdkill);
+EFFECT(mdauto);
+EFFECT(destruct);
+// For mode sounds, specific "mdstun", "mdkill", and "mdauto" may be used.
+// If just a single "mode" sound for all switches exists, that will be used.
+// If no mode sounds exist in the font, a talkie version will speak the mode on switching.
+#ifndef PROPS_DUAL_PROP_H
+EFFECT(battery);    // for EFFECT_BATTERY_LEVEL
+EFFECT(vmbegin);    // for Begin Volume Menu
+EFFECT(vmend);      // for End Volume Menu
+EFFECT(volup);      // for increse volume
+EFFECT(voldown);    // for decrease volume
+EFFECT(volmin);     // for minimum volume reached
+EFFECT(volmax);     // for maximum volume reached
+EFFECT(quote);      // for playing quotes
+#endif
+
+class BlasterBC : public PROP_INHERIT_PREFIX PropBase {
```

Comment:
It can't inherit from the blaster prop?
(If not that's ok, but it should be considered if it will reduce code duplication)

Reply:
I am unclear on this. Since only one blaster prop file is included at a time, the other isn't even compiled, no?
Isn't it code substitution as opposed to code duplication?
I can imagine if there were a blaster_base.h maybe...

Reply:
Well, here's a reverse diff reporting what's the same between blaster.h and my prop. 271 same lines.
Is this good/worth making a blaster_base for?
[blaster_props_shared_code.txt](https://github.com/profezzorn/ProffieOS/files/12334977/blaster_props_shared_code.txt)

Reply:
yes, absolutely.

Reply:
Well, i'm modifying it now, assuming we're leaving it named blaster.h, not making a new file blaster_base.h, right?

Reply:
ok. inheritance complete. How about now?